products/apigateway/api.yaml

```diff
@@ -113,7 +113,6 @@ objects:
         name: apiConfigId
         description: |
           Identifier to assign to the API Config. Must be unique within scope of the parent resource(api).
-        required: true
```

Comment:
Can we use `exactly_one_of` instead, which makes one of the two required?

Reply:
I tried but it doesn't work. It generates an empty ExactlyOneOf config. Have a look at the generated diff from 9a2ce6c

Reply:
The other thing is that the custom encoder generates a random name if the api_config_id or api_config_id_prefix is not supplied.

Reply:
Oh! In that case we don't want the additional validation. Sorry- missed that it also took a value when neither was specified!